### PR TITLE
Lock TS deps to exact versions

### PR DESCRIPTION
**Pull request by banool**
## Description
In this PR we:
- Use exact versions in all the package.json files. This is what we do in our Rust deps, we specify exactly right down to the patch version. This is a good way to avoid an unintentional package upgrade leading to a supply chain attack. At the very least we should be using `~`, but I still think that's unsafe. If we have lock files, we clearly care about what versions we're using, let's be explicit about it at the top level (and keep lockfiles for locking transitive deps).
- Update the deps such that they use the versions we previously had in the lock files, since the locked version was naturally often higher than the "careted" version in the package.json file.
- Update the JS SDK examples to use `aptos` version `1.3.12`.
- Remove `tslib` as a dep from the JS examples for the TS SDK, since it shouldn't be necessary now that we're using `tsup`.
- Check in a lockfile for `ecosystem/platform`.
- Build `ecosystem/node-checker/ts-client` with `tsup`.

Some oddities:
- Apparently the community platform was only using version `1.2.0` of the `aptos` TS package. We're up to `1.3.12` now, so this seems odd to me / it's unlikely this works at all? I must be missing something. I've added the web platform folks to comment + review the changes in their dir.

## Test Plan
CI.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4206)

### CODE REVIEW

File: ecosystem/node-checker/ts-client/tsup.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { defineConfig } from "tsup";
+
+export default defineConfig({
+  entry: ["src/index.ts"],
+  format: ["esm", "cjs"],
+  splitting: false,
+  sourcemap: true,
+  clean: true,
+  dts: true,
+  globalName: "aptosSDK",
```
Comment (jjleng) [inline ecosystem/node-checker/ts-client/tsup.config.js:10]: This is not `aptosSDK`.

**Review by jjleng — COMMENTED** [review]
No need to pin devDependencies.

**Review by rajkaramchedu — APPROVED** [review]

File: ecosystem/platform/server/package.json [hunk 2]
```diff
@@ -2,17 +2,17 @@
...
-    "tailwindcss": "^3.1.0"
+    "@hotwired/stimulus": "3.0.1",
+    "@hotwired/turbo-rails": "7.1.3",
+    "@sentry/browser": "7.12.1",
+    "@tailwindcss/forms": "0.5.2",
+    "@tailwindcss/typography": "0.5.2",
+    "aptos": "1.2.0",
+    "autoprefixer": "10.4.7",
+    "dotenv": "16.0.2",
+    "esbuild": "0.14.39",
```
Comment (zacharydenton) [inline ecosystem/platform/server/package.json:13]: It appears to have downgraded esbuild.

**Review by zacharydenton — APPROVED** [review]

File: ecosystem/platform/server/package.json (hunk 2 above)
Comment (banool) [reply ecosystem/platform/server/package.json:13]: Thanks for catching this.

**Comment by banool** [comment]
Okay I did another pass and made sure everything looks good in the lock files. I'll land this now.

**Pull request merged**